tests/data/test_maestro.py

```diff
@@ -0,0 +1,192 @@
+
+RESOURCES_PATH = pathlib.Path(__file__).parent.parent / "resources"
+MAESTRO_TEST_DATA_PATH = RESOURCES_PATH / "data" / "maestro"
+
+TRAIN_TRACK_ID = "2004/MIDI-Unprocessed_SMF_05_R1_2004_01_ORIG_MID--AUDIO_05_R1_2004_03_Track03_wav"
+VALID_TRACK_ID = "2004/MIDI-Unprocessed_SMF_02_R1_2004_01-05_ORIG_MID--AUDIO_02_R1_2004_06_Track06_wav"
+TEST_TRACK_ID = "2004/MIDI-Unprocessed_SMF_02_R1_2004_01-05_ORIG_MID--AUDIO_02_R1_2004_08_Track08_wav"
+GT_15M_TRACK_ID = "2004/MIDI-Unprocessed_SMF_02_R1_2004_01-05_ORIG_MID--AUDIO_02_R1_2004_05_Track05_wav"
+
+
+def create_mock_wav(output_fpath: str, duration_min: int) -> None:
+    duration_seconds = duration_min * 60
+    sample_rate = 44100
+    n_channels = 2  # Stereo
+    sampwidth = 2  # 2 bytes per sample (16-bit audio)
+
+    # Generate a silent audio data array
+    num_samples = duration_seconds * sample_rate
+    audio_data = np.zeros((num_samples, n_channels), dtype=np.int16)
+
+    # Create the WAV file
+    with wave.open(str(output_fpath), "w") as wav_file:
+        wav_file.setnchannels(n_channels)
+        wav_file.setsampwidth(sampwidth)
+        wav_file.setframerate(sample_rate)
+        wav_file.writeframes(audio_data.tobytes())
+
+    logging.info(f"Mock {duration_min}-minute WAV file '{output_fpath}' created successfully.")
+
+
+def create_mock_midi(output_fpath: str) -> None:
+    # Create a new MIDI file with one track
+    mid = MidiFile()
+    track = MidiTrack()
+    mid.tracks.append(track)
+
+    # Define a sequence of notes (time, type, note, velocity)
+    notes = [
+        (0, "note_on", 60, 64),  # C4
+        (500, "note_off", 60, 64),
+        (0, "note_on", 62, 64),  # D4
+        (500, "note_off", 62, 64),
+    ]
+
+    # Add the notes to the track
+    for time, type, note, velocity in notes:
+        track.append(Message(type, note=note, velocity=velocity, time=time))
+
+    # Save the MIDI file
+    mid.save(output_fpath)
+
+    logging.info(f"Mock MIDI file '{output_fpath}' created successfully.")
+
+
+def test_maestro_to_tf_example(tmpdir: str) -> None:
+    mock_maestro_home = pathlib.Path(tmpdir) / "maestro"
+    mock_maestro_ext = mock_maestro_home / "2004"
+    mock_maestro_ext.mkdir(parents=True, exist_ok=True)
+
+    create_mock_wav(str(mock_maestro_ext / (TRAIN_TRACK_ID.split("/")[1] + ".wav")), 3)
+    create_mock_midi(str(mock_maestro_ext / (TRAIN_TRACK_ID.split("/")[1] + ".midi")))
+
+    output_dir = pathlib.Path(tmpdir) / "outputs"
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    input_data: List[str] = [TRAIN_TRACK_ID]
+    with TestPipeline() as p:
+        (
+            p
+            | "Create PCollection of track IDs" >> beam.Create([input_data])
+            | "Create tf.Example" >> beam.ParDo(MaestroToTfExample(str(mock_maestro_home), download=False))
+            | "Write to tfrecord" >> beam.ParDo(WriteBatchToTfRecord(str(output_dir)))
+        )
+
+    assert len(os.listdir(str(output_dir))) == 1
```

Comment:
I'd make os.listdir() a variable if you're gonna call it this many times. Unsure if you need the `str` coercion